tensorflow/core/common_runtime/zen_layout_pass.cc

```diff
@@ -0,0 +1,1196 @@
+    output_nodes->push_back(NodeBuilder::NodeOut(n, slot));
+    (*input_idx)++;
+    list_length--;
+  }
+}
+
+Status ZenLayoutRewritePass::ZenOpNodeRewrite(
+    std::unique_ptr<Graph> *g, Node *orig_node,
+    const ZenOpRewriteRecord *rewrite_record,
+    std::pair<bool, bool> reorder_flags) {
+  DCHECK_NE(rewrite_record, nullptr);
+  DCHECK_NE(orig_node, nullptr);
+
+  Node *new_node = nullptr;
+  std::vector<string> fused_ops = {};
+  int num_data_inputs = orig_node->in_edges().size();
+  for (const Edge *e : orig_node->in_edges()) {
+    if (e->IsControlEdge()) {
+      num_data_inputs--;
+    }
+  }
+
+  gtl::InlinedVector<Node *, 4> control_edges;
+  gtl::InlinedVector<std::pair<Node *, int>, 4> inputs(num_data_inputs);
+  FillInputs(orig_node, &control_edges, &inputs);
+
+  NodeBuilder nb(orig_node->name().c_str(),
+                 rewrite_record->zen_op_name.c_str());
+
+  nb.Device(orig_node->def().device());
+  TF_RETURN_IF_ERROR(CopyInputs(orig_node, inputs, &nb));
+  rewrite_record->update_zen_op_attr(const_cast<const Node *>(orig_node), &nb);
+
+  nb.Attr("reorder_before", reorder_flags.first);
+  nb.Attr("reorder_after", reorder_flags.second);
+  nb.Attr("in_links", IncomingEdgeCount(orig_node));
+  nb.Attr("out_links", OutgoingEdgeCount(orig_node));
+  nb.Attr("reset", IsLastZenNode(g, orig_node));
+
+  // Add/Update Fused Op Attribute.
+  if (orig_node->type_string() == "_ZenFusedConv2D" ||
+      orig_node->type_string() == "_FusedConv2D" ||
+      orig_node->type_string() == "_FusedDepthwiseConv2dNative" ||
+      orig_node->type_string() == "_ZenFusedDepthwiseConv2dNative") {
+    TF_CHECK_OK(GetNodeAttr(orig_node->def(), "fused_ops", &fused_ops));
+    if (FuseConv2DBiasRelu(g, orig_node, "Relu")) {
+      if (fused_ops.size() == 1) {
+        fused_ops.push_back("Relu");
+      }
+    }
+    if (FuseConv2DBiasRelu(g, orig_node, "Relu6")) {
+      if (fused_ops.size() == 1) {
+        fused_ops.push_back("Relu6");
+      }
+    }
+    nb.Attr("fused_ops", fused_ops);
+  }
+  TF_RETURN_IF_ERROR(nb.Finalize(&**g, &new_node));
+
+  std::unordered_set<Node *> unique_node;
+  for (const Edge *e : orig_node->in_edges()) {
+    if (e->IsControlEdge()) {
+      auto result = unique_node.insert(e->src());
+      if (result.second) {
+        (*g)->AddControlEdge(e->src(), new_node, true);
+      }
+    }
+  }
+  unique_node.clear();
+
+  for (const Edge *e : orig_node->out_edges()) {
+    if (e->IsControlEdge()) {
+      auto result = unique_node.insert(e->dst());
+      if (result.second) {
+        (*g)->AddControlEdge(new_node, e->dst(), true);
+      }
+    } else {
+      auto result =
+          (*g)->AddEdge(new_node, e->src_output(), e->dst(), e->dst_input());
+      DCHECK_NE(result, nullptr);
+    }
+  }
+  new_node->set_assigned_device_name(orig_node->assigned_device_name());
+  (*g)->RemoveNode(orig_node);
+
+  return OkStatus();
+}
+
+// 'nodes' is a vector of Zen nodes marked for rewrite to update reorder flags.
```

Comment:
Actually, let's move this to the function documentation in line 329, e.g.,
```c++
@input nodes - A vector of Zen nodes marked for rewrite ...
```

Reply:
Moved the explanation to function declaration. Thank you!